sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/InvitationService.java

```diff
@@ -213,9 +223,15 @@ private void checkExisting(String email) {
 
     private void sendEmail(Invitation invitation) {
         IEmailMessage userInviteEmail = LDAP.equals(invitation.getSource())
-                ? new UserInviteEmail(invitation.getToken(), zafiraLogoURL, urlResolver.buildWebURL())
-                : new UserInviteLdapEmail(invitation.getToken(), zafiraLogoURL, urlResolver.buildWebURL());
+                ? new UserInviteEmail(invitation.getInvitationUrl(), zafiraLogoURL, urlResolver.buildWebURL())
+                : new UserInviteLdapEmail(invitation.getInvitationUrl(), zafiraLogoURL, urlResolver.buildWebURL());
         emailService.sendEmail(userInviteEmail, invitation.getEmail());
     }
 
+    private void insertInvitationUrl(Invitation invitation) {
```

Comment:
method signature looks a little bit sketchy... it kinda fills invitation dto depending on some condition rather than insert something somewhere. How about instead of this private method use something like this:
`invitation.setInvitationUrl(invitation != null ? urlResolver.buildInvitationUrl(invitation.getToken()) : null);`
that way you won't deal with mutational void methods and logic will be more explicit.